api/src/main/java/org/apache/iceberg/ExpireSnapshots.java

```diff
@@ -40,11 +40,13 @@
 public interface ExpireSnapshots extends PendingUpdate<List<Snapshot>> {
 
   /**
-   * Expires a specific {@link Snapshot} identified by id.
+   * Expires a specific {@link Snapshot} identified by id. This API isn't compatible with
+   * cherry-pick functionality (issue #744), and hence marked Deprecated.
    *
    * @param snapshotId long id of the snapshot to expire
    * @return this for method chaining
    */
+  @Deprecated
```

Comment:
Can you move this to a separate PR? I don't think we need to update RemoveSnapshots and deprecate a method at the same time.

Reply:
Also can we possibly provide a message to use an alternate API for deprecation. Something like - 
```java
@Deprecate("Use `expireOlderThan()` method instead")
```